bond_app/sam_api.py

```diff
@@ -12,14 +12,14 @@ def user_info(self, access_token):
         """
         Calls sam GET /register/user/v1 api
         :param access_token: oauth access token
-        :return: dict with userSubjectId and userEmail keys or None if user does not exist in sam
+        :return: dict with userSubjectId and userEmail keys or else raises an exception if user does not exist in sam
         """
         headers = {'Authorization': 'Bearer ' + access_token}
         result = requests.get(url=self.base_url + '/register/user/v1?userDetailsOnly=true', headers=headers)
         if result.status_code == 200:
             return json.loads(result.content)["userInfo"]
         elif result.status_code == 404:
-            return None
+            raise exceptions.Unauthorized("user not found in sam")
```

Comment:
I disagree with this change. The sam_api module should reflect the sam api and let the caller determine what to do with the response. The appropriate return value for a 404 is None and the caller should raise the appropriate exception or handle it

Reply:
That being said, if the status is anything other than `2xx` here, we should add a logging statement to log the response from Sam.

Reply:
sg, reverted fence_token_vending and added checks to bond.py to handle None return case.
Also added logging here.